### PR TITLE
Common Lisp: A Contender.

### PR DESCRIPTION
**Pull request by pnathan**
The Common Lisp system used is SBCL, which has gotten the bulk of the performance improvements for a long time.

The code has been tuned, but plausibly there might be other speedups possible.

~~There might also be some remarkable optimizations going on under the hood as well.~~ 

Update: Inner loop was optimized to do 10x less work due to programmer derp.

```
 $ time ./c/code 22
1059323

real    0m1.978s
user    0m1.966s
sys     0m0.012s
(base) 2024-11-25 18:20:23  ~/external/source/languages/loops:
 $ time ./common-lisp/code 22
9473
real    0m6.700s
user    0m6.692s
sys     0m0.008s

### CODE REVIEW

File: loops/common-lisp/code.lisp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        (a (make-array 10000   :element-type '(unsigned-byte 32) ))
+        (u (parse-integer (second *posix-argv*)) ))
+    (declare
+     (type (mod 10000) r)
+     (type (unsigned-byte 32) u)
+     (type (simple-array (unsigned-byte 32) (10000)) a))
+
+
+    (loop for i of-type (mod 10001) from 0 below 10000 do
+      (loop for j of-type (mod 10001) from 0 below 10000 do
```
Comment (klirix) [inline loops/common-lisp/code.lisp:15]: shouldn't the [inner loop be 100k](https://github.com/bddicken/languages/blob/7d2d3ee6691335456a499975cfb2b1e9fe4937da/loops/c/code.c#L10), not 10k?🤔

Comment (pnathan) [reply loops/common-lisp/code.lisp:15]: quite so. A dazzling feat of 0s.

**Comment by bddicken** [comment]
Excellent, thanks for submitting this.

**Comment by PEZ** [comment]
Will we have Common Lisp back on the charts now that there is a new benchmark runner?

* #371

**Pull request merged**